README.md

```diff
@@ -1,1110 +1,8 @@
-<!-- START doctoc generated TOC please keep comment here to allow auto update -->
```

Comment:
@Bobgy I cleaned up the README and divided this into two parts, oss-test-infra and optional-test-infra.

Let me know if you want us to better organize the documentation

Reply:
project root level does need a README.md. I would suggest to move some content out of the root README.md instead of deleting it

Reply:
I didn't delete those doc but moved it into gcp_README.md. 

Maybe I can add some common content back to README.md, what do you think?

Reply:
sounds good. Seems some contents are general information here.

Reply:
Moved contents back to root README